src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,399 @@
+  camera.restoreTransformation();
+
+  // Color texture ONLY, NOT for depth
+  if (flags_ & (Flag::BuildMipMap | Flag::ColorTexture)) {
+    textures_[TextureType::Color]->generateMipmap();
+  }
+}
+
+void CubeMap::loadTexture(TextureType type,
+                          const std::string& imageFilePrefix,
+                          const std::string& imageFileExtension) {
+  auto loadImporter = [&](const std::string& plugin) {
+    Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> manager;
+    Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+        manager.loadAndInstantiate(plugin);
+    CORRADE_ASSERT(importer,
+                   "CubeMap::loadTexture(): cannot initialize" << plugin, );
+
+    imageImporterManger_.set<Mn::Trade::AbstractImporter>(
+        plugin, importer.release(), Mn::ResourceDataState::Final,
+        Mn::ResourcePolicy::Manual);
+  };
+
+  std::string plugin;
+  if (imageFileExtension == "png") {
+    plugin = "PngImporter";
+  } else if (imageFileExtension == "jpg") {
+    plugin = "JpegImporter";
+  } else if (imageFileExtension == "hdr") {
+    plugin = "HdrImporter";
+  } else {
+    LOG(INFO) << "CubeMap::loadTexture(): cannot support image type "
+              << imageFileExtension;
+    CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+  }
+
+  Mn::Resource<Mn::Trade::AbstractImporter> importer =
+      imageImporterManger_.get<Mn::Trade::AbstractImporter>(plugin);
+  // create one if there is no such importer
+  if (!importer) {
+    loadImporter(plugin);
+  }
+  CORRADE_INTERNAL_ASSERT(importer);
+
+  std::string coordStrings[6] = {".+X", ".-X", ".+Y", ".-Y", ".+Z", ".-Z"};
+  int imageSize = 0;
+  // set images
+  Mn::GL::CubeMapTexture* texture;
+  switch (type) {
+    case TextureType::Color:
+      texture = textures_[TextureType::Color].get();
+      break;
+
+    case TextureType::Depth:
+      texture = textures_[TextureType::Depth].get();
+      break;
+
+    default:
+      CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+      break;
+  }
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    // open image file
+    std::string filename =
+        imageFilePrefix + getTextureTypeFilenameString(type) +
+        coordStrings[iFace] + "." + std::string{imageFileExtension};
+    importer->openFile(filename);
+    Cr::Containers::Optional<Mn::Trade::ImageData2D> image =
+        importer->image2D(0);
+
+    // sanity checks
+    CORRADE_INTERNAL_ASSERT(image);
+    Mn::Vector2i size = image->size();
+    CORRADE_ASSERT(
+        size.x() == size.y(),
+        " CubeMap::loadTexture(): each texture image must be a square.", );
+    if (iFace == 0) {
+      imageSize = size.x();
+      reset(imageSize);
+    } else {
+      CORRADE_ASSERT(size.x() == imageSize,
+                     " CubeMap::loadTexture(): texture images must have the "
+                     "same size.", );
+    }
+
+    texture->setSubImage(convertFaceIndexToCubeMapCoordinate(iFace), 0, {},
+                         *image);
```

Comment:
Did you test with HDR images? Because I expect this to not work, as the HDR images give you a RGB but the texture is just a single channel.

Maybe GL does some conversion under the hood, but I wouldn't bet on that.

Reply:
NO, I have not tested it with HDR yet! I was about to ask how to implemented in the correct way.
You mentioned to me on the slack channel that somehow I need to convert it (from RGB separate channels to a singl channel). Any code snippet to demo how to do it would be appreciated here.

Reply:
... or we can go with OpenEXR and don't bother with limitations of the HDR format :)

It's not hard, basically you convert a tightly-packed stream of RGBRGBRGB... data to RRRR..., throwing away the two other channels.